app/scripts/controllers/deploymentConfig.js

```diff
@@ -87,6 +95,33 @@ angular.module('openshiftConsole')
             $scope.deploymentConfig = deploymentConfig;
             updateHPAWarnings();
             ImageStreamResolver.fetchReferencedImageStreamImages([deploymentConfig.spec.template], $scope.imagesByDockerReference, $scope.imageStreamImageRefByDockerReference, context);
+            copyDeploymentConfigAndEnsureEnv(deploymentConfig);
+            $scope.saveEnvVars = function() {
+              // copy again to clean off any of angulars $$ vars from the template
```

Comment:
Might have been overly cautious doing the extra copy w/o first trying it.  I'd say remove & test it. If it still works, lets drop the extra angular.copy().  

https://github.com/angular/angular.js/blob/master/src/ng/http.js
Looks like `$http` uses `.toJSON` (via `.serializeValue`)